### PR TITLE
fix(tester.py): Set networktopologyreplication for system_auth

### PR DESCRIPTION
**Pull request by aleksbykov**
Replication strategy for system_auth should be set for number of nodes in cluster and with appropriate Replication Strategy. See: https://docs.scylladb.com/stable/operating-scylla/security/authentication.html

task: #6084 

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/tester.py [hunk 1]
```diff
@@ -1421,7 +1446,7 @@ def get_cluster_k8s_local_kind_cluster(self):
         else:
             self.monitors = NoMonitorSet()
 
-    @staticmethod
+    @ staticmethod
```
Comment (soyacz) [inline sdcm/tester.py:1449]: what are these spaces between?

File: sdcm/tester.py [hunk 2]
```diff
@@ -856,7 +857,28 @@ def setUp(self):  # pylint: disable=too-many-branches,too-many-statements
...
+        dc_name = self.db_cluster.get_nodetool_info(nodes_by_region[region][0])['Data Center']
+        for keyspace in keyspaces:
+            replication_strategy = ReplicationStrategy.get(node, keyspace)
+            if not isinstance(replication_strategy, SimpleReplicationStrategy):
+                # no need to switch as already is NetworkTopology
+                continue
+            self.log.info("Switching replication strategy to Network for '%s' keyspace", keyspace)
+            if keyspace == "system_auth" and replication_strategy.replication_factor != len(nodes_by_region[region]):
+                self.log.warning("system_auth keyspace is not replicated on all nodes (%s)",
+                                 replication_strategy.replication_factor / len(nodes_by_region[region]))
```
Comment (soyacz) [inline sdcm/tester.py:875]: instead of warning it should set rf to len(nodes) and do it for each region.

File: sdcm/tester.py [hunk 3]
```diff
@@ -856,7 +857,28 @@ def setUp(self):  # pylint: disable=too-many-branches,too-many-statements
                 db_cluster.validate_seeds_on_all_nodes()
                 validate_raft_on_nodes(nodes=db_cluster.nodes)
 
+    def _switch_to_network_replication_strategy(self, keyspaces: List[str]) -> None:
```
Comment (soyacz) [inline sdcm/tester.py:860]: This case is specific and I wouldn't generalize it - better create method like `switch_system_auth_to_network_topology_strategy` and focus on it.

File: sdcm/tester.py [hunk 4]
```diff
@@ -856,7 +857,28 @@ def setUp(self):  # pylint: disable=too-many-branches,too-many-statements
...
+                # no need to switch as already is NetworkTopology
+                continue
+            self.log.info("Switching replication strategy to Network for '%s' keyspace", keyspace)
+            if keyspace == "system_auth" and replication_strategy.replication_factor != len(nodes_by_region[region]):
+                self.log.warning("system_auth keyspace is not replicated on all nodes (%s)",
+                                 replication_strategy.replication_factor / len(nodes_by_region[region]))
+            network_replication = NetworkTopologyReplicationStrategy(
+                **{dc_name: replication_strategy.replication_factor})
+            network_replication.apply(node, keyspace)
+
```
Comment (roydahan) [inline sdcm/tester.py:879]: Let's run "Desc keyspace" for the changed keyspace, so we will have the output in the log.

**Comment by aleksbykov** [comment]
Check with multidc job, result for system_auth with the fix:

CREATE KEYSPACE system_auth WITH replication = {'class': 'NetworkTopologyStrategy', 'eu-west-2scylla_node_west': '5', 'eu-westscylla_node_west': '5'}  AND durable_writes = true;


for single dc:

CREATE KEYSPACE system_auth WITH replication = {'class': 'NetworkTopologyStrategy', 'eu-west': '4'}  AND durable_writes = true;

File: sdcm/tester.py (hunk 1 above)
Comment (aleksbykov) [reply sdcm/tester.py]: Fixed, looks like IDE bug

File: sdcm/tester.py (hunk 2 above)
Comment (aleksbykov) [reply sdcm/tester.py]: Removed and leave simple code to change the Replication class

File: sdcm/tester.py (hunk 3 above)
Comment (aleksbykov) [reply sdcm/tester.py]: Just change logic of previous function

File: sdcm/tester.py (hunk 4 above)
Comment (aleksbykov) [reply sdcm/tester.py]: Added appropriate code.

**Comment by aleksbykov** [comment]
Reopened it, because close by mistake

File: sdcm/tester.py [hunk 5]
```diff
@@ -865,19 +866,21 @@ def set_system_auth_rf(self, db_cluster=None):
             # TODO: move this skip to siren-tools when possible
             self.log.warning("Skipping this function due this job run from Siren cloud!")
             return
-        # change RF of system_auth
-        system_auth_rf = self.params.get('system_auth_rf')
```
Comment (fruch) [inline sdcm/tester.py:869]: If we aren't using this one anymore, we should remove it from all configuration files (and from sct_config.py)

Comment (aleksbykov) [reply sdcm/tester.py:869]: removed

**Review by fruch — APPROVED** [review]
LGTM

**Comment by roydahan** [comment]
What about taking care of increasing and decreasing the RF every time we add/remove node?

**Comment by fgelcer** [comment]
i see that [this artifacts on Azure](https://jenkins.scylladb.com/job/enterprise-2023.1/job/artifacts/job/artifacts-azure-image-test/29/) failed while running the code introduced here... can please someone help me look into it?

**Comment by fgelcer** [comment]
> i see that [this artifacts on Azure](https://jenkins.scylladb.com/job/enterprise-2023.1/job/artifacts/job/artifacts-azure-image-test/29/) failed while running the code introduced here... can please someone help me look into it?

@aleksbykov @soyacz

**Comment by fgelcer** [comment]
> > i see that [this artifacts on Azure](https://jenkins.scylladb.com/job/enterprise-2023.1/job/artifacts/job/artifacts-azure-image-test/29/) failed while running the code introduced here... can please someone help me look into it?
> 
> @aleksbykov @soyacz

well, all [of these](https://argus.scylladb.com/workspace?state=WyI2NjIwOGI2Ny0yNTBjLTRmOTMtODhjOS0wODE3MjYzMDNlODQiLCI4MWVhZTIwZC05M2Y4LTRhY2MtODdiOC0zMzNlYmI1YzZhMjgiLCI5NTlkNGQwNS1iNDQ4LTQ4YmEtODMxYi03MTk3MjRkMWM3NjQiLCJhZjJiNDZjYi1mNTBmLTQzNTQtYjI3MC00NDZkMjUxZTYwMjgiLCJjYzAzMTYzYy1jNGVmLTRmMDItYjE4NS1iN2NiZWRiZTY0MTAiLCJmNzNlNTE2My01NTYzLTQ2NzEtOGZhOS02ZTcyMTE2MDg1OWIiXQ) also failed with the same problem...
could it be that, `cqlsh` is not in the path when we tried to run this command?

**Pull request merged**